httpclient5/src/main/java/org/apache/hc/client5/http/entity/compress/CompressingFactory.java

```diff
@@ -0,0 +1,277 @@
+    private final Map<String, String> formattedNameCache = new ConcurrentHashMap<>();
+
+    /**
+     * Returns a set of available input stream compression providers.
+     *
+     * @return a set of available input stream compression providers in lowercase.
+     */
+    public Set<String> getAvailableInputProviders() {
+        return inputProvidersCache.updateAndGet(existing -> existing != null ? existing : fetchAvailableInputProviders());
+    }
+
+    /**
+     * Returns a set of available output stream compression providers.
+     *
+     * @return a set of available output stream compression providers in lowercase.
+     */
+    public Set<String> getAvailableOutputProviders() {
+        return outputProvidersCache.updateAndGet(existing -> existing != null ? existing : fetchAvailableOutputProviders());
+    }
+
+    /**
+     * Returns the formatted name of the provided compression format.
+     * <p>
+     * If the provided name matches an alias (e.g., "gzip" or "x-gzip"), the method will return the standard name.
+     * </p>
+     *
+     * @param name the compression format name.
+     * @return the formatted name, or the original name if no alias is found.
+     * @throws IllegalArgumentException if the name is null or empty.
+     */
+    public String getFormattedName(final String name) {
+        if (name == null || name.isEmpty()) {
+            LOG.warn("Compression name is null or empty");
+            return null;
+        }
+        final String lowerCaseName = name.toLowerCase(Locale.ROOT);
+        return formattedNameCache.computeIfAbsent(lowerCaseName, key -> {
+            if ("gzip".equals(key) || "x-gzip".equals(key)) {
+                return "gz";
+            } else if ("compress".equals(key)) {
+                return "z";
+            }
+            return key;
+        });
+    }
+
+    /**
+     * Creates an input stream for the specified compression format and decompresses the provided input stream.
+     * <p>
+     * This method uses the specified compression name to decompress the input stream and supports the "noWrap" option
+     * for deflate streams.
+     * </p>
+     *
+     * @param name        the compression format.
+     * @param inputStream the input stream to decompress.
+     * @param noWrap      if true, disables the zlib header and trailer for deflate streams.
+     * @return the decompressed input stream, or the original input stream if the format is not supported.
+     */
+    public InputStream getCompressorInputStream(final String name, final InputStream inputStream, final boolean noWrap) throws CompressorException {
+        Args.notNull(inputStream, "InputStream");
+        Args.notNull(name, "name");
+        final String formattedName = getFormattedName(name);
+        return isSupported(formattedName, false)
+                ? createCompressorInputStream(formattedName, inputStream, noWrap)
+                : inputStream;
```

Comment:
The Javadoc and implementation don't match:
- Javadoc states decompression is performed
- The code doesn't decompress if the format isn't supported.

Either the Javadoc needs updating, or, more likely, we should throw an exception indicating the operation is not supported for that format type name.

I must say I find the API and Javadoc confusing, we are creating a "Compressor" input stream in order to "decompress" with a "compression" format! What?

- Maybe calling the API `getDecompressorInputStream` would help because it "decompresses". I expect a "Compressor" to compress, not the other way around.
- Maybe using the term "format type name" instead "compression format" would remove the need for the reader to constantly parse "compress" vs. "decompress" in the text. 
- Maybe only using one term "decompress" or "compress" for one method (Javadoc and code) would alleviate the issue.

Reply:
Thanks, @garydgregory . Agreed—the Javadoc and implementation don’t match. We’ll:

- Throw an exception on unsupported formats instead of returning the original stream, aligning with Javadoc expectations.

- Rename to getDecompressorInputStream to reflect decompression.
- Standardize to “format type” instead of “compression format” to clarify.
-
These changes should make both the API and docs clearer. Thanks for flagging this.

Reply:
HI @garydgregory 
Can you do another pass?? 
Thank you.